includes/class-sensei-course.php

```diff
@@ -1264,6 +1264,62 @@ public function course_lessons( $course_id = 0, $post_status = 'publish', $field
 		return $lessons;
 
 	}
+	/**
+	 * Get lessons by courses.
+	 *
+	 * @since  4.3.0
+	 * @access public
+	 *
+	 * @param array $courses_ids The list of courses ids.
+	 * @return array[] lists of lessons by courses.
+	 */
+	public function get_lessons_grouped_by_courses( $courses_ids ) {
+		if ( ! is_array( $courses_ids ) || empty( $courses_ids ) ) {
+			return [];
+		}
+		global $wpdb;
+
+		// phpcs:ignore WordPress.DB.DirectDatabaseQuery.DirectQuery, WordPress.DB.DirectDatabaseQuery.NoCaching -- Safe direct sql.
+		return $wpdb->get_results(
+			// phpcs:ignore
+			$wpdb->prepare(
+				"SELECT pm.meta_value as course_id, pm.post_id as lesson_id
+				FROM {$wpdb->postmeta} pm
+				WHERE pm.meta_value IN (%1s)" // phpcs:ignore WordPress.DB.PreparedSQLPlaceholders.UnquotedComplexPlaceholder -- no need for quoting.
+				. " AND pm.meta_key = '_lesson_course'
+				GROUP BY pm.post_id",
+				implode( ',', $courses_ids )
+			)
+		);
+	}
+
+	/**
+	 * Get students by courses.
+	 *
+	 * @since  4.3.0
+	 * @access public
+	 *
+	 * @param array $courses_ids The list of courses ids.
+	 * @return array[] lists of students based on the courses list.
+	 */
+	public function get_students_grouped_by_courses( $courses_ids ) {
```

Comment:
I would not add it until we decide on the project level which direction we want to go, since we do not have it almost anywhere else.

Reply:
I think the only reason these are sparse throughout the code is because support for type declarations wasn't added until PHP 7.0, and we had to support PHP 5.6 in Sensei for quite awhile. As long as a PHP feature has support in at least version 7.0, we should feel free to use it.